signcrypt_seal.go

```diff
@@ -69,6 +69,7 @@ func (sss *signcryptSealStream) signcryptBytes(b []byte) error {
 
 	nonce := nonceForChunkSigncryption(sss.numBlocks)
 
+	// TODO: Encode payload chunk.
```

Comment:
Another reason I want to include associated data at the encryption layer: A similar problem applies to authenticating packet against the header hash. We could go either way with the final packet flag, because that one does actually get written into every packet, and it's just a question of where. But we don't have a choice with the header hash, since that one's not supposed to be transported in the packet at all. We shouldn't include it in the secretbox (that would create duplication-and-so-possible-inconsistency apart from wasting space), so our only option is to include it as associated data. Given that we have to add an AD mechanism for that bit, we might as well use the same mechanism for the final packet flag?